src/main/java/org/testng/TestNG.java

```diff
@@ -875,9 +875,7 @@ private void initializeCommandLineSuitesGroups() {
     }
   }
   private void addReporter(Class<? extends IReporter> r) {
-    if (! m_reporters.contains(r)) {
-      m_reporters.add(ClassHelper.newInstance(r));
-    }
+    m_reporters.add(ClassHelper.newInstance(r));
```

Comment:
`m_reporters.contains(r)` always return false since m_reporters is a `Map<IReporter>` where r is `Class<? extends Reporter>`